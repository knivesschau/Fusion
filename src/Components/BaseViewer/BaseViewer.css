/* mobile styles first, min 320px */
.View_Base {
    text-align: center;
    background-color: rgb(245, 245, 245, .9);
    border: 1px solid rgb(128, 128, 128);
    border-radius: 2em; 
    margin: 1.5em; 
}

#choose-recipe, #pick-another {
    cursor: pointer;
    display: inline-block;
    box-sizing: border-box;
    background-color: #A9C09B;
    text-transform: uppercase;
    color: black;
    font-size: .65rem;
    font-weight: bold;
    margin-left: auto;
    margin-right: auto;
    margin: .5em;
    margin-top: 1.5em;
    padding: 5px 10px;
    border: 1px solid black;
    border-radius: .75em;
    transition: all .15s;
}

#choose-recipe:hover, #pick-another:hover {
    text-shadow: 0 0 32px rgba(255, 255, 255, 1);
    color: white;
    border-color: white;
}

#choose-recipe:active, #pick-another:active {
    background-color: #881600;
}

@media screen and (min-device-width: 375px) {
    #choose-recipe, #pick-another {
        font-size: .85rem;
    }
}

/* tablet and desktop styles */ 
@media screen and (min-width: 768px) {
    .View_Base {
        margin-left: 5em;
        margin-right: 5em;
        border-radius: 3em;
        margin-top: 2.5em;
    }

    #choose-recipe, #pick-another {
        font-size: 1.10rem;
        padding-top: .75em;
        padding-bottom: .75em;
        margin-right: -.5em;
        margin-left: 2em;
    }
}

@media screen and (min-width: 1200px) {
    .View_Base {
        margin-left: 15em;
        margin-right: 15em;
        margin-top: 3.5em;
    }

    #starting-cuisine {
        font-size: 1.75rem;
    }

    #choose-recipe, #pick-another {
        font-size: 1.25rem;
    }
}

@media screen and (min-width: 1600px) {
    .View_Base {
        margin-left: 30em;
        margin-right: 30em;
    }
}

@media only screen and (min-width: 2000px) {
    .View_Base {
        margin-left: 35em;
        margin-right: 35em;
    }
}