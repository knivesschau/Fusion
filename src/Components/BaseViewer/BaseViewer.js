import React, { Component } from 'react';
import fusionContext from '../../fusionContext';
import BaseRecipe from '../BaseRecipe/BaseRecipe'
import './BaseViewer.css';

export default class BaseView extends Component {
    static defaultProps = {
        match: {
            params: {}
        }
    };

    handleCancelClick = () => {
        this.props.history.push(`/starter-recipes`);
    };

    static contextType = fusionContext;

    render() {

        const {bases=[]} = this.context; 
        const {recipe_id} = this.props.match.params; 
        const baseRecipe = bases.find(base => base.recipe_id === parseInt(recipe_id)) || {};

        return (
            <section className="View_Base">
                {/* pass all starter recipe data to BaseRecipe Component via props */}
                <BaseRecipe
                    recipe_id={baseRecipe.recipe_id}
                    base_name={baseRecipe.base_name}
                    cuisine_name={baseRecipe.cuisine_name}
                    ingredients={baseRecipe.ingredients}
                    steps={baseRecipe.steps}
                />
                
            <button type="button" id="choose-recipe" onClick={e => this.props.history.push(`/fuse/${recipe_id}`)}>Fuse Recipe</button>
            <button type="button" id="pick-another" onClick={this.handleCancelClick}>Pick Another Recipe</button>

            </section>
        );
    };
};