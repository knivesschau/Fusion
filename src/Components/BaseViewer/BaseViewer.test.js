import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import BaseViewer from './BaseViewer';

describe ('BaseViewer Component', () => {
    it ('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BaseViewer/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('Renders the complete BaseViewer Component and all nested Components', () => {
        const wrapper = shallow(<BaseViewer/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});