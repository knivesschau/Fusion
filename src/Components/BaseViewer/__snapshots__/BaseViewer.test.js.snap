// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseViewer Component Renders the complete BaseViewer Component and all nested Components 1`] = `
<section
  className="View_Base"
>
  <button
    id="choose-recipe"
    onClick={[Function]}
    type="button"
  >
    Fuse Recipe
  </button>
  <button
    id="pick-another"
    onClick={[Function]}
    type="button"
  >
    Pick Another Recipe
  </button>
  <BaseRecipe />
</section>
`;
